client/components/domains/register-domain-step/index.jsx

```diff
@@ -844,22 +844,22 @@ class RegisterDomainStep extends Component {
 
 	fetchDomainPricePromise = ( domain ) => {
 		return new Promise( ( resolve ) => {
-			wpcom.undocumented().getDomainPrice( domain, ( serverError, result ) => {
-				if ( serverError ) {
+			wpcom.req
+				.get( `/domains/${ encodeURIComponent( domain ) }/price` )
+				.then( ( data ) => {
+					resolve( {
+						pending: false,
+						is_premium: data.is_premium,
+						cost: data.cost,
+						is_price_limit_exceeded: data?.is_price_limit_exceeded,
```

Comment:
No need for `?.` as the other `data` accesses are not guarded either.

Reply:
Good call, cleaned up in https://github.com/Automattic/wp-calypso/pull/58902/commits/36d9c3431f373e408f4ceb80e99d289742340b4e